common/src/main/java/com/wynntils/features/user/BeaconBeamFeature.java

```diff
@@ -32,14 +33,16 @@ public List<Class<? extends Model>> getModelDependencies() {
 
     @SubscribeEvent
     public void onRenderLevelLast(RenderTileLevelLastEvent event) {
-        if (CompassModel.getCompassLocation().isEmpty()) return;
+        Optional<Location> compassLocation = CompassModel.getCompassLocation();
+        if (compassLocation.isEmpty()) return;
 
         PoseStack poseStack = event.getPoseStack();
         MultiBufferSource.BufferSource bufferSource =
                 McUtils.mc().renderBuffers().bufferSource();
 
         Vec3 camera = event.getCamera().getPosition();
-        Location location = CompassModel.getCompassLocation().get();
+        Location location = compassLocation.get();
+        location.set(location.x, 0, location.z);
```

Comment:
Will this modify the underlying representation, or do we return a copy?

Reply:
It modifies the representation. Why?

Reply:
I'm not entirely happy about the compass location API. It feels too open and leaky. I realize there's a difference between locations that are fixed (like NPCs) and that can change (like compass point), but the API should be more restrictive here, I think. I noticed you and Incomplete did some stuff on static vs dynamic poi's that I've not had the time to look into detail. Maybe it's the beginning on getting this more strict.

Also, we should perhaps consider having a location representation where we can actually flag the Y value as "don't know/don't care".